vizro-core/examples/_dev/app.py

```diff
@@ -1,31 +1,100 @@
-"""Example to show dashboard configuration."""
+"""Dev app to try things out."""
```

Comment:
In just a few minutes, I was able to create a button component to show the number of rows that are are displayed in the Graph component above. 
Here are some of my observations along with the code example I made:

- Figure (reactive button component in the example) is set as `export_data` target and it works well. Parametrised and filtered button's `data_frame` that goes into the `my_export_data_button_figure` is exported properly (I guess that's something we wanted to achieve). 
- Figure (reactive button component in the example) as  the `filter_interaction` target also works well.
- However, I haven't been able to configure the app so that the Figure (button) can trigger the `export_data` action. I tried with `class MyFigureButton(vm.Figure):` but it raises the same exception as if I try to inherit a new class from any of Graph/Table/AgGrid. Probably the only way would be to write the MyFigureButton class from scratch but it doesn't seem like something we want to recommend. So:
  - What is suggested way to achieve this?
  - Should we expose `actions` together with the `_input_component_id` arguments in Figure, something similar we did for AgGrid and Table? 


```python
import vizro.models as vm
import vizro.plotly.express as px
from vizro import Vizro
from vizro.models.types import capture
import dash_bootstrap_components as dbc
from vizro.actions import export_data
from vizro.managers import data_manager


@capture("figure")
def my_export_data_button_figure(data_frame, show_how_many_rows: bool = True, **kwargs):
    return dbc.Button(
        children=f'Export {len(data_frame)} rows' if show_how_many_rows else "Export data",
        **kwargs
    )


def load_first_n_data(first_n_points=150):
    return px.data.iris().head(first_n_points)


data_manager["iris"] = load_first_n_data


page_two = vm.Page(
    title="Export N-row Data",
    components=[
        vm.Graph(id="my_graph", figure=px.scatter("iris", x="sepal_width", y="sepal_length", color="species")),
        vm.Figure(
            id="my_button",
            figure=my_export_data_button_figure("iris"),
        ),
        vm.Button(text="Vizro export button", actions=[vm.Action(function=export_data())])
    ],
    controls=[
        vm.Filter(column="species"),
        vm.Parameter(
            targets=[
                "my_graph.data_frame.first_n_points",
                "my_button.data_frame.first_n_points",
            ],
            selector=vm.Slider(
                id="first_n_points",
                min=0,
                max=150,
                step=1,
                marks={i: str(i) for i in range(0, 151, 15)},
                value=150,
            ),
        )
    ],
)


dashboard = vm.Dashboard(pages=[page_two])

if __name__ == "__main__":
    Vizro().build(dashboard).run()
```

Reply:
Hey @petar-qb ,

Thanks for checking this so thoroughly! 🚀🕵️‍♂️ 

I've left the `action` argument out for now, as we first wanted to see how people might want to use `Figure` as a source of action. I've agreed with @antonymilne that the `Figure` should be enabled as the output of any action, but for it to be the source, the user would currently have to encode it. The code for any kind of filter interaction, e.g., would also have to live with the model itself, so for that, they already have to change the source code of `Figure.` 

Copying/pasting this much code might be suboptimal for now, but at least it's there 😄 If we notice that many people want to do that, we can always expose it later after we have a better understanding.

I've added the open tasks on the `Figure` here and your comment above: https://github.com/McK-Internal/vizro-internal/issues/1019.

Reply:
Thanks for summarising all Figure TODOs in one place, I like it. 

As you linked this thread in the "Enhance Figure" issue, just to add that all stuff like: `_filter_interaction_input`, `_filter_interaction`, `_action_validator_factory` argument, `_output_component_property` argument, `_input_component_id` and similar will potentially be configured over the upcoming `"actions_info"` feature.

Reply:
Not sure if you tested this one:

```python
vm.Parameter(targets=["my_button.show_how_many_rows"], selector=vm.RadioItems(
            options=[{"label": "Show number of rows", "value": True},
                     {"label": "Don't show number of rows", "value": False}], value=True)),
```
but it also works well, so parmetrising the `CapturedCallable` not just the data.

If you say the `filter_interaction` works well - and we know that `Figure` as action source is not intended for now, then I think all is tested!

Super cool!